Dockerfile

```diff
@@ -2,16 +2,19 @@ FROM python:3-slim AS builder
 
 RUN mkdir /src
 COPY . /src/
+ENV VIRTUAL_ENV=/opt/venv
+RUN python -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 RUN pip install --no-cache-dir --upgrade pip setuptools wheel \
     # Install build tools to compile dependencies that don't have prebuilt wheels
     && apt update && apt install -y git build-essential \
     && cd /src \
-    && pip install --user --no-cache-dir .[colorama,d]
+    && pip install --no-cache-dir .[colorama,d]
 
 FROM python:3-slim
 
 # copy only Python packages to limit the image size
-COPY --from=builder /root/.local /root/.local
-ENV PATH=/root/.local/bin:$PATH
+COPY --from=builder /opt/venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
```

Comment:
Will this make the PATH persistent like this?

Reply:
Yes, it will prepend the PATH for all users. Here's a couple examples: 

```zsh
# test output as host user
❯ docker run --user $(id -u):$(id -g) nborges/black:pr which black
/opt/venv/bin/black

# test output as root
❯ docker run --user 0:0 nborges/black:pr which black
/opt/venv/bin/black
```